dockstore-integration-testing/src/test/java/io/dockstore/client/cli/BasicIT.java

```diff
@@ -1492,6 +1491,25 @@ public void testGettingSourceFilesForTag() {
         if (!throwsError) {
             Assert.fail("Should not be able to grab sourcefile for a version not belonging to a tool");
         }
+
+        // check that sourcefiles can't be viewed by another user if they aren't published
+        throwsError = false;
+        final io.dockstore.openapi.client.ApiClient user2OpenAPIWebClient = getOpenAPIWebClient(USER_2_USERNAME, testingPostgres);
+        io.dockstore.openapi.client.api.ContainertagsApi user2toolTagsApi = new io.dockstore.openapi.client.api.ContainertagsApi(user2OpenAPIWebClient);
+        try {
+            sourceFiles = user2toolTagsApi.getTagsSourcefiles(tool.getId(), tag.getId(), null);
```

Comment:
I think the other pattern (where we put the assertFail() right after this line) is cleaner since there's less code and no extra boolean.